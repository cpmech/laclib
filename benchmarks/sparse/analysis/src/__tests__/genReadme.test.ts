import { genReadme } from '../genReadme';
import { genHtml } from '../genHtml';

describe('genReadme', () => {
  it('generates readme filw with bfwb62', () => {
    const readme = genReadme(
      ['bfwb62'],
      ['open', 'intel'],
      ['seq_omp#', 'mpi1_omp#', 'mpi#', 'mpi#_omp1', 'mpi#_omp#'],
    );
    expect(readme).toMatchSnapshot();

    // debugging
    const filepath = '/tmp/laclib/genReadme.test.html';
    genHtml(filepath, 'laclib genReadme test', readme, true);
    console.log(`file <${filepath}> written`);
  });
});
