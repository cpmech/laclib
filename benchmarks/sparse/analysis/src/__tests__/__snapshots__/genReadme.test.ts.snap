// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genReadme generates readme filw with bfwb62 1`] = `
"# Benchmarks using the code for sparse matrices

The code here tests the perfomance of the MUMPS Sparse Solver.

## Bai / bfwb62 matrix

Reference: https://sparse.tamu.edu/Bai/bfwb62

* Symmetric matrix
* Dimension = 62
* Number of non-zero values = 202
* L-infinity norm of the matrix = 1.000e-4

_results with \\"metis\\" ordering:_

<table style=\\"border: 1px solid #979797; border-collapse: collapse;\\">
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">bfwb62</td>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">case</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">NP or NT 1</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">NP or NT 2</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">NP or NT 3</th>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">NP or NT 4</th>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"10\\">Analyze</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">open_seq_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">463.178µs</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">709.094µs</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">732.382µs</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">726.987µs</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">intel_seq_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">2.26055ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">9.37454ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">8.10867ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.7606ms</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">open_mpi1_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.7058ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.80039ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.37454ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3.98261ms</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">intel_mpi1_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6.92038ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3.21484ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">7.29034ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">7.64976ms</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">open_mpi#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.66629ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">8.2673ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">5.41377ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6.06771ms</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">intel_mpi#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">5.23238ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">8.30394ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">14.0877ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">19.079ms</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">open_mpi#_omp1</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.7058ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">8.60622ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">7.66633ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">5.14488ms</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">intel_mpi#_omp1</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6.92038ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">11.2438ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">8.49033ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">9.49058ms</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">open_mpi#_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.7058ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">1.80039ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">8.60622ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">7.50725ms</td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">intel_mpi#_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">6.92038ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">3.21484ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">11.2438ms</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\">11.5611ms</td>
  </tr>
  <tr>
    <th style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\" rowspan=\\"10\\">Factorize</th>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">open_seq_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_seq_omp1.txt\\">376.599µs</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_seq_omp2.txt\\">8.76785ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_seq_omp3.txt\\">28.5722ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_seq_omp4.txt\\">9.00163ms</a></td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">intel_seq_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_seq_omp1.txt\\">22.5825ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_seq_omp2.txt\\">4.8315ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_seq_omp3.txt\\">4.69297ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_seq_omp4.txt\\">11.9224ms</a></td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">open_mpi1_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi1_omp1.txt\\">1.64748ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi1_omp2.txt\\">2.08928ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi1_omp3.txt\\">1.20363ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi1_omp4.txt\\">6.424ms</a></td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">intel_mpi1_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi1_omp1.txt\\">10.297ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi1_omp2.txt\\">6.58081ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi1_omp3.txt\\">8.18585ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi1_omp4.txt\\">14.3465ms</a></td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">open_mpi#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi1.txt\\">1.28033ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi2.txt\\">1.6484ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi3.txt\\">926.544µs</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi4.txt\\">1.05918ms</a></td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">intel_mpi#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi1.txt\\">7.48603ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi2.txt\\">3.62667ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi3.txt\\">3.69428ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi4.txt\\">23.6338ms</a></td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">open_mpi#_omp1</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi1_omp1.txt\\">1.64748ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi2_omp1.txt\\">656.11µs</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi3_omp1.txt\\">1.33194ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi4_omp1.txt\\">998.586µs</a></td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">intel_mpi#_omp1</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi1_omp1.txt\\">10.297ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi2_omp1.txt\\">4.06645ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi3_omp1.txt\\">3.24646ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi4_omp1.txt\\">3.20821ms</a></td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">open_mpi#_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi1_omp1.txt\\">1.64748ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi1_omp2.txt\\">2.08928ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi2_omp1.txt\\">656.11µs</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/open/mumps_bfwb62_metis_open_mpi2_omp2.txt\\">506.321µs</a></td>
  </tr>
  <tr>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: right; padding: 5px;\\">intel_mpi#_omp#</td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi1_omp1.txt\\">10.297ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi1_omp2.txt\\">6.58081ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi2_omp1.txt\\">4.06645ms</a></td>
    <td style=\\"border: 1px solid #979797; border-collapse: collapse; text-align: center; padding: 5px;\\"><a href=\\"https://github.com/cpmech/laclib/blob/main/benchmarks/sparse/results/intel/mumps_bfwb62_metis_intel_mpi2_omp2.txt\\">3.36536ms</a></td>
  </tr>
</table>


"
`;
